spec/models/provider_spec.rb

```diff
@@ -731,4 +721,46 @@
       end
     end
   end
+
+  describe "#search_by_code_or_name" do
+    let!(:matching_provider) { create(:provider, provider_code: "ABC", provider_name: "Dave's Searches") }
+    let!(:non_matching_provider) { create(:provider) }
+
+    subject { described_class.search_by_code_or_name(search_term) }
+
+    context "with an exactly matching code" do
+      let(:search_term) { "ABC" }
+      it { is_expected.to contain_exactly(matching_provider) }
+    end
+
+    context "with an exactly matching name" do
+      let(:search_term) { "Dave's Searches" }
+      it { is_expected.to contain_exactly(matching_provider) }
+    end
+
+    context "with unicode in the name" do
+      let(:search_term) { "Dave’s Searches" }
+      it { is_expected.to contain_exactly(matching_provider) }
+    end
+
+    context "with extra spaces in the name" do
+      let(:search_term) { "Dave's  Searches" }
+      it { is_expected.to contain_exactly(matching_provider) }
+    end
+
+    context "with non matching case code" do
+      let(:search_term) { "abc" }
+      it { is_expected.to contain_exactly(matching_provider) }
+    end
+
+    context "with non matching case name" do
+      let(:search_term) { "dave's searches" }
+      it { is_expected.to contain_exactly(matching_provider) }
+    end
+
+    context "with partial search term" do
+      let(:search_term) { "dave" }
```

Comment:
Just out of curiosity search terms as dive or ave?

Reply:
It seems to only match complete words at the moment but we can expand that in config.

So with a provider name Boys' School

matches:

* school boys
* boys school
* boys’   school

doesn't match:

* bys school

Reply:
This isn't true now... see below